src/Stratis.Bitcoin.Features.Consensus/PowConsensusValidator.cs

```diff
@@ -725,7 +914,7 @@ private void MerkleComputation(List<uint256> leaves, ref uint256 root, ref bool
             while (count < leaves.Count)
             {
                 uint256 h = leaves[(int)count];
-                bool matchh = count == branchpos;
+                bool match = false;
```

Comment:
Change of logic has to be reviewed carefully

Reply:
yes and same below on new line 938

Reply:
btw this PR is not ready for review i think, ...

Reply:
Yep, I'll ping when it is completed. There are still a few things I have to check and improve.

Reply:
>Change of logic has to be reviewed carefully

Totally agreed. 

Regarding that change: 
branchpos and pbranch were there in order to achieve a small performance acceleration in some rare cases (so we don't have to rebuild whole tree every time) but we never used it and always filled this params with -1 and null so  
`count == branchpos` statement is always false. 

I've checked if old and new version of ComputeMerkleRoot produces the same result and it does. 
Also if new implementation would calculate merkle root differently consensus would stop advancing, so before pushing anything I'm checking if consensus advances.

Reply:
> I've checked if old and new version of ComputeMerkleRoot produces the same result and it does.

i'd recommend to create a set of tests for that that should pass before the change and after the change as well
or are there such tests implemented already?

Reply:
Good idea. 
Though I guess it will be a good idea to write tests for merkle root calculation after this PR so we don't make dangershony wait too long till this one is merged. 

>or are there such tests implemented already?

No, there are no tests involving this merkle root calculation code.

Reply:
> so before pushing anything I'm checking if consensus advances

Great yes thats one way to do it, tests will be also really nice to have (double check NBitcoin does not  have some already)

Reply:
>btw this PR is not ready for review i think, 

Ok @Aprogiena I will leave most of this to you to review unless you need my help let me know.